packages/ui/react-ui-editor/src/components/TextEditor/TextEditor.stories.tsx

```diff
@@ -15,75 +13,52 @@ import { fixedInsetFlexLayout, getSize, groupSurface, inputSurface, mx } from '@
 import { withTheme } from '@dxos/storybook-utils';
 
 import { TextEditor } from './TextEditor';
-import { createHyperlinkTooltip, hyperlink } from './extensions';
+import { createHyperlinkTooltip, hyperlinkDecoration } from './extensions';
 import { useTextModel } from '../../model';
 
-export const nameRegex = /\{([\w_]+)}/;
-
-/**
- * Simple Monaco language extension.
- * https://github.com/codemirror/stream-parser/blob/main/test/test-stream-parser.ts
- */
-export const promptLanguage = StreamLanguage.define<{ count: number }>({
-  startState: () => ({ count: 0 }),
-
-  token: (stream, state) => {
-    state.count++;
-    if (stream.eatSpace()) {
-      return null;
-    }
-    if (stream.match(/^(<\w+>)/)) {
-      return 'tagName';
-    }
-    stream.next();
-    return null;
-  },
-});
-
-/**
- * https://codemirror.net/examples/styling
- * https://lezer.codemirror.net/docs/ref/#highlight
- */
-export const promptHighlightStyles = HighlightStyle.define([
-  {
-    tag: tags.tagName,
-    class: mx(
-      'mr-1 p-1',
-      'bg-neutral-100 dark:bg-neutral-900 text-black dark:text-white font-mono text-sm',
-      'rounded border border-neutral-300 dark:border-neutral-700',
-    ),
-  },
-]);
-
-export const defaultHyperLinkTooltip = createHyperlinkTooltip((el, url) => {
-  const web = new URL(url);
-  createRoot(el).render(
-    <StrictMode>
-      <div className='flex gap-1 items-center'>
-        <ArrowCircleUp className={mx(getSize(6), 'text-blue-500')} />
-        <p className='pr-1'>{web.origin}</p>
-      </div>
-    </StrictMode>,
-  );
-});
+const text = [
+  '',
+  '',
+  'This is all about [DXOS](https://dxos.org); read more [here](https://docs.dxos.org/guide/getting-started.html).',
+  '',
+  '',
+].join('\n');
 
-const Story = () => {
-  const [item] = useState({
-    text: new TextObject(
-      '\n\n\nThis is all about [DXOS](https://dxos.org); read more [here](https://docs.dxos.org/guide/getting-started.html).\n\n\n',
-    ),
+const hyperLinkTooltip = () =>
+  createHyperlinkTooltip((el, url) => {
+    const web = new URL(url);
+    createRoot(el).render(
+      <StrictMode>
+        <div className='flex gap-1 items-center'>
+          <ArrowCircleUp className={mx(getSize(6), 'text-blue-500')} />
```

Comment:
This should probably be `ArrowSquareOut` if this does what I think it does.

Reply:
i don't feel strongly, although i interpret ArrowSquareOut as expanding a node -- e.g., i would use this for stack sections to go into full screen mode, rather than commanding the browser to popout a link.